### PR TITLE
Add cluster key statistics in block meta

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add cluster key statistics in block meta

## Changelog

- New Feature

## Related Issues

Fixes #issue

### CODE REVIEW

File: query/src/storages/fuse/fuse_table.rs [hunk 1]
```diff
@@ -152,7 +152,7 @@ impl Table for FuseTable {
         ctx: Arc<QueryContext>,
         stream: SendableDataBlockStream,
     ) -> Result<SendableDataBlockStream> {
-        let log_entry_stream = self.append_trunks(ctx, stream).await?;
+        let log_entry_stream = self.append_chunks(ctx, stream).await?;
```
Comment (dantengsky) [inline query/src/storages/fuse/fuse_table.rs:155]: thanks :)

File: query/src/storages/fuse/meta/common.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ pub struct Statistics {
     pub compressed_byte_size: u64,
 
     pub col_stats: HashMap<ColumnId, ColumnStatistics>,
+    pub cluster_stats: HashMap<ColumnId, ColumnStatistics>,
```
Comment (dantengsky) [inline query/src/storages/fuse/meta/common.rs:37]: Just a reminder:

we need a `#[serde(default)]` attr for the new field `cluster_stats` to keep backward compatible in the final PR

File: query/src/storages/fuse/meta/v1/segment.rs [hunk 3]
```diff
@@ -48,6 +48,7 @@ pub struct BlockMeta {
     pub file_size: u64,
     pub col_stats: HashMap<ColumnId, ColumnStatistics>,
     pub col_metas: HashMap<ColumnId, ColumnMeta>,
+    pub cluster_stats: HashMap<ColumnId, ColumnStatistics>,
```
Comment (dantengsky) [inline query/src/storages/fuse/meta/v1/segment.rs:51]: Just a reminder:

we need a `#[serde(default)]` attr for the new field `cluster_stats` to keep backward compatible in the final PR

File: query/src/storages/fuse/fuse_table.rs (hunk 1 above)
Comment (zhyass) [reply query/src/storages/fuse/fuse_table.rs:155]: > thanks :)

https://github.com/datafuselabs/databend/pull/5194

Comment (dantengsky) [reply query/src/storages/fuse/fuse_table.rs:155]: Got it

**Pull request closed without merging**